store/json.go

```diff
@@ -153,21 +147,20 @@ func (kvs *jsonFileStore) Lock(block bool) error {
 			break
 		}
 
-		if !block || i == lockMaxRetries {
-			return ErrStoreLocked
+		if !block {
```

Comment:
Again not your code. But a better pattern would be to have:

`func TryLock() bool` -> Non-Blocking return value indicates `wasLocked`
`func Lock()` -> Blocking
`func Unlock()` -> Same for both